posix/fcntl.c

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <openenclave/corelibc/errno.h>
+#include <openenclave/corelibc/limits.h>
+#include <openenclave/internal/posix/device.h>
+#include <openenclave/internal/posix/fdtable.h>
+#include <openenclave/internal/posix/raise.h>
+#include <openenclave/internal/trace.h>
+#include "mount.h"
+#include "posix_t.h"
+
+int __oe_fcntl(int fd, int cmd, uint64_t arg)
+{
+    int ret = -1;
+    oe_device_t* device;
+
+    if (!(device = oe_fdtable_get(fd, OE_DEVICE_TYPE_NONE)))
+        OE_RAISE_ERRNO(OE_EINVAL);
+
+    ret = OE_CALL_BASE(fcntl, device, cmd, arg);
+
+done:
+    return ret;
+}
+
+int oe_open(const char* pathname, int flags, oe_mode_t mode)
+{
+    int ret = -1;
+    int fd;
+    oe_device_t* fs;
+    oe_device_t* file;
+    char filepath[OE_PATH_MAX] = {0};
+
+    if (!(fs = oe_mount_resolve(pathname, filepath)))
+        OE_RAISE_ERRNO(oe_errno);
+
+    if (!(file = OE_CALL_FS(open, fs, filepath, flags, mode)))
```

Comment:
In case of an error in a the line at 41, should file be closed?

Reply:
Thanks Radhika. Good find! I fixed this one an a few others.